### PR TITLE
net-misc/r8125-9.008.00: fixed build with Linux Kernel 5.17.x

### PR DESCRIPTION
**Pull request by Karlson2k**
No revbump, no need to rebuild if module works.

Closes: https://bugs.gentoo.org/839282

### CODE REVIEW

File: net-misc/r8125/files/r8125-9.008.00-linux-5.17.patch [hunk 1]
```diff
@@ -0,0 +1,112 @@
+Added compatibility with Linux Kernel 5.17+.
```
Comment (thesamesam) [inline net-misc/r8125/files/r8125-9.008.00-linux-5.17.patch:1]: Please include at the top of the patch:
- link to where it's from / submitted to
- link to relevant Gentoo bug

Comment (Karlson2k) [reply net-misc/r8125/files/r8125-9.008.00-linux-5.17.patch:1]: The patch was designed by myself, based on my analysis. There is nowhere to submit it, Realtek doesn't have any public repo. Later I discovered that my patch is almost identical to [the relevant patch for R8168](https://github.com/gentoo/gentoo/commit/13afe46537e99157756939d331bebed49169dfc6), which is a good sign. 

I'll add the info.

**Comment by Karlson2k** [comment]
Added info to the patch.
Added second commit with bump of EAPI.

**Review by juippis — APPROVED** [review]
Yeah I guess this will be handled by `@module-rebuild`. Just FYI I couldn't build net-misc/r8125-9.007.01::gentoo due to `cc1: some warnings being treated as errors` - something you may want to fix in future. net-misc/r8125-9.008.00 was fine though.

**Comment by Karlson2k** [comment]
> Yeah I guess this will be handled by `@module-rebuild`. Just FYI I couldn't build net-misc/r8125-9.007.01::gentoo due to `cc1: some warnings being treated as errors` - something you may want to fix in future. net-misc/r8125-9.008.00 was fine though.

@juippis `net-misc/r8125-9.007.01` needs to be patched to work with 5.17+ kernels. My patch doesn't work with 9.007.01 and I'm not sure that it worth to invest my time to patch the old version. Probably it's better just to drop it.

**Comment by juippis** [comment]
What was the point for that EAPI bump then? :) but I'll drop it, thanks for notifying.

**Pull request closed without merging**